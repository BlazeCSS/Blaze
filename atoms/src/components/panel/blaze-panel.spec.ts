import { flush, render } from '@stencil/core/testing';
import { Panel } from './blaze-panel';

describe('Panel', () => {
  it('should build', () => {
    expect(new Panel()).toBeTruthy();
  });

  let element;

  const snapIt = (name, html) => {
    it(name, async () => {
      element = await render({
        components: [Panel],
        html
      });
      await flush(element);

      expect(element).toMatchSnapshot();
    });
  };

  snapIt(
    'renders correctly with no props',
    '<blaze-panel>default</blaze-panel>'
  );

  snapIt(
    'renders correctly with height set',
    '<blaze-panel height="350">default</blaze-panel>'
  );
});
