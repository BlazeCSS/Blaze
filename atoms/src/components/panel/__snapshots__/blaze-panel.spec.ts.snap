// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly with height set 1`] = `
<blaze-panel
  class="hydrated"
  height="350"
>
  <div
    class="o-panel-container"
    style="height: 350px;"
  >
    <div
      class="o-panel"
    >
      <!---->
      default
    </div>
  </div>
</blaze-panel>
`;

exports[`Panel renders correctly with no props 1`] = `
<blaze-panel
  class="hydrated"
>
  <div
    class="o-panel-container"
    style="height: 0px;"
  >
    <div
      class="o-panel"
    >
      <!---->
      default
    </div>
  </div>
</blaze-panel>
`;
